html_tag_collector/collector.py

```diff
@@ -36,7 +36,7 @@ def process_urls(manager_list, render_javascript):
     """
     df = manager_list[0]
     urls = df.select(pl.col("url")).rows()
-    new_urls = ["https://" + url[0] if not url[0].startswith("http") else url[0] for url in urls]
+    new_urls = ["https://" + url[0] if (url[0] is not None and not url[0].startswith("http")) else url[0] for url in urls]
```

Comment:
Nit here: As a single line, the logic is hard to grasp -- list comprehension with two if conditionals and string concatenation is a lot to put on one line. Recommend breaking it up into multiple lines. For example:

```python
new_urls = []  
for url in urls:
    if url[0] is not None and not url[0].startswith("http"):
        new_urls.append("https://" + url[0])
    else:
        new_urls.append(url[0])
```

Reply:
good call, done!